clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -972,16 +975,28 @@ private Record nextFetchedRecord() {
         private List<ConsumerRecord<K, V>> fetchRecords(int maxRecords) {
             if (isFetched)
                 return Collections.emptyList();
+            if (nextInlineException != null) {
```

Comment:
@hachikuji I would like to make a few clarifications:

- The example you described is wrong. After we fetch the record at offset 1, the position is at 2, not 1.
- After we catch an error caused by an invalid record, the position is the offset of that invalid record. This is how the new consumer has worked up until now.
- This patch has not changed behavior regarding the whether consumer fetches from the current position. As of current Kafka implementation, regardless of whether it is by design or accident, consumer will skip the invalid message after throwing exception. The purpose of this patch is only to make sure that consumer doesn't skip valid messages before the invalid message. I thought you agreed that this can be addressed in a separate patch.